AntonBabunin/src/artcode/shop/Bill.java

```diff
@@ -0,0 +1,100 @@
+package artcode.shop;
+
+
+public class Bill {
+    private Product[] products;
+    private double amountPrice;
+    private String seller;
+    private String time;
+    private boolean isClosed = false;
+
+    public Bill(){
+
+    }
+
+//    public Bill(Product[] products, double amountPrice, String seller, String time) {
+//        this.products = products;
+//        this.amountPrice = amountPrice;
+//        this.seller = seller;
+//        this.time = time;
+//    }
+
+    public Product[] getProducts() {
+        return products;
+    }
+
+    public void setProducts() {
+        this.products = new Product[20];
+    }
+
+    public double getAmountPrice() {
+        return amountPrice;
+    }
+
+    private void calculate() {
+        double amountPrice = 0.0;
+        for (Product prod : products) {
+            if (prod != null) {
+                amountPrice += prod.getPrice();
+            }
+        }
+        this.amountPrice = amountPrice;
+    }
+
+    public String getSeller() {
+        return seller;
+    }
+
+    public void setSeller(String seller) {
+        this.seller = seller;
+    }
+
+    public String getTime() {
+        return time;
+    }
+
+    public void setTime(String time) {
+        this.time = time;
+    }
+
+    public void addProduct (Product product) {
+        if (this.getProducts() == null || this.getProducts().equals(null)){
+            setProducts();
+            if (product != null && !isClosed()) {
+                addProductToArr(product);
+            }
+        }   else {
+            addProductToArr(product);
+        }
+    }
+
+    private void addProductToArr (Product product) {
+        for (int i = 0; i < this.products.length; i++) {
```

Comment:
I prefer to keep some counter, that will know about last free position, otherwise, you probably will iterate over that 1 000  000 elements until you will find null. As a result, you will slow you app

Reply:
Added two private int fields at Bill: firstFreePositionAtProducts and nextFreePositionAtProducts and some logic in metods addProduct  and  delFromBill (Product product);